### ISSUE TITLE
swap! causes too much recursion

### ISSUE THREAD

**Issue by LiFlash** [reagent-project/reagent-cursor#15]
When swapping a cursor like so
``(swap! my-cursor update-in [:my-path] into ["some" "data"])``
I get a 'too much recursion' in the js console when accessing the atom again.

When using 
``(reset! my-cursor (update-in @my-cursor [:my-path] into ["some" "data"]))``

instead, everything is fine. I'm a clojure(script) noob and don't have a clue if I am the reason for this or not...

Thanks and greetings, Joe

**Comment by Frozenlock** [comment]
Everything works here...

```clj
(def aaa (atom {:a {:b {:c {:d 1}}}}))
;=> #<Atom: {:a {:b {:c {:d 1}}}}>

(def ccc (c/cur aaa [:a :b]))
;=> #<Cursor: {:c {:d 1}} @[:a :b]>

;; here's your example command:
(swap! ccc update-in [:my-path] into ["some" "data"])
;=> {:c {:d 1}, :my-path ("data" "some")}

;; we can even do it again...
(swap! ccc update-in [:my-path] into ["some" "data"])
;=> {:c {:d 1}, :my-path ("data" "some" "data" "some")}
```

Are you using reagent.cursor provided by this library, or using the one still in reagent.core?

**Comment by Frozenlock** [comment]
Closed as discussed in #14
